### PR TITLE
fix: Use actual EIP4361 ABNF grammar

### LINKED ISSUE #16: The siwe message produced by ceramic-cacao package is different from the one produced by spruceid/siwe
minimum code to reproduce

```js
import { SiweMessage as CeramicSiweMessage } from "ceramic-cacao";
import { SiweMessage } from "siwe";

const params = {
  domain: "service.org",
  address: "0xdb46d1dc155634fbc732f92e853b10b288ad5a1d",
  statement:
    "I accept the ServiceOrg Terms of Service: https://service.org/tos",
  uri: "did:key:z6MkrBdNdwUPnXDVD1DCxedzVVBpaGi8aSmoXFAeKNgtAer8",
  version: "1",
  nonce: "32891757",
  issuedAt: "2021-09-30T16:25:24.000Z",
  chainId: "1",
  resources: [
    "ceramic://k2t6wyfsu4pg040dpjpbla1ybxof65baldb7fvmeam4m3n71q0w1nslz609u2d",
  ],
};

const siweMessage = new SiweMessage(params);
const ceramicSiweMessage = new CeramicSiweMessage(params);

const message = siweMessage.toMessage();
const ceramicMessage = ceramicSiweMessage.toMessage();

console.log(message === ceramicSiweMessage); // false
```

### PR DESCRIPTION
**Pull request by ukstv**
Fix #16 

How it was verified against what is reported in #16 :
I created a js package (referenced as _testing package_ below) that uses the code from the issue, with a slight change to compare actual strings:
```ts
console.log(message === ceramicMessage);
```
It reported `false`. Then I built a code in this PR, copied it to corresponding file in `node_modules` of my testing package.
It started to report `true`.

### CODE REVIEW

**Comment by ukstv** [comment]
Oh, it is not done yet. Backwards compatibility is a bitch. Older cacaos in use still rely on the old format of the string for verification. Have to add an if switch to verify both variants of ordering.

File: src/siwx/siwx.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import type { Cacao } from '../cacao.js'
 import { ParsedMessage as ABNFParsedMessage } from '../abnf.js'
 import { AccountId, ChainId } from 'caip'
 
+const CHAIN_ID_REORG_DATE = new Date('2022-08-25').valueOf()
```
Comment (oed) [inline src/siwx/siwx.ts:5]: Maybe make this like 14 days from now? Tight timeline otherwise.

**Review by oed — COMMENTED** [review]
Can we have a tests for backward compatibility?

**Comment by ukstv** [comment]
> Can we have a tests for backward compatibility? 

Uhums, with the current PR the compatibility is not clear. Gonna make it clear.

**Comment by ukstv** [comment]
Closing in favour of https://github.com/ceramicnetwork/cacao/pull/20

**Pull request closed without merging**